logging/src/main/java/com/nightlynexus/retrofit/logging/LoggingCallAdapterFactory.java

```diff
@@ -26,12 +29,23 @@ public final class LoggingCallAdapterFactory extends CallAdapter.Factory {
    * It is an error to mutate the call from these methods.
    */
   public interface Logger {
-    <T> void onResponse(Call<T> call, Response<T> response);
-
-    <T> void onFailure(Call<T> call, Throwable t);
+    Object UNBUILT_REQUEST_BODY = new Object();
```

Comment:
Or maybe there should be a different callback for the case where building the request fails.